system/Database/DataConverter/Cast/DatetimeCast.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Database\DataConverter\Cast;
+
+use CodeIgniter\I18n\Time;
+
+/**
+ * Class DatetimeCast
+ *
+ * PHP: Time <--> DB column: datetime
+ *
+ * @extends BaseCast<Time, string, mixed>
+ */
+class DatetimeCast extends BaseCast
+{
+    /**
+     * {@inheritDoc}
+     */
+    public static function fromDatabase(mixed $value, array $params = []): Time
+    {
+        if (! is_string($value)) {
+            self::invalidTypeValueError($value);
+        }
+
+        return Time::parse($value);
```

Comment:
I know this is hearkening back to a recent conversation, which I don't think we resolved, but... should this be the equivalent of `createFromFormat()` instead? I'm a little worried about `parse()` accepting too broad of input.

Reply:
Agreed.
Changed to `createFromFormat()`: 024c5547bb2a1ac2ad5ae46031a14fe826b99a4a